src/panels/lovelace/common/validate-condition.ts

```diff
@@ -1,27 +1,46 @@
-import { UNAVAILABLE } from "../../../data/entity";
 import { HomeAssistant } from "../../../types";
+import { evaluateFilter, validFilter } from "./evaluate-filter";
 
 export interface Condition {
   entity: string;
-  state?: string;
-  state_not?: string;
+  state?: string | string[] | any[];
+  state_not?: string | string[];
+  state_filter?: Array<{ key: string } | string>;
 }
 
 export function checkConditionsMet(
   conditions: Condition[],
   hass: HomeAssistant
 ): boolean {
   return conditions.every((c) => {
-    const state = hass.states[c.entity]
-      ? hass!.states[c.entity].state
-      : UNAVAILABLE;
+    const stateObj = hass.states[c.entity];
+    if (!stateObj) return false;
 
-    return c.state ? state === c.state : state !== c.state_not;
+    if (c.state_filter) {
+      if (!c.state_filter.some((f) => evaluateFilter(stateObj, f)))
+        return false;
+    }
+
+    if (c.state) {
+      if (c.state !== stateObj.state) return false;
+    }
+
+    if (c.state_not) {
+      if (c.state_not === stateObj.state) return false;
+    }
+    return true;
   });
 }
 
 export function validateConditionalConfig(conditions: Condition[]): boolean {
-  return conditions.every(
-    (c) => (c.entity && (c.state || c.state_not)) as unknown as boolean
-  );
+  return conditions.every((c) => {
+    if (!c.entity) return false;
+    if (c.state && typeof c.state !== "string") return false;
+    if (c.state_not && typeof c.state_not !== "string") return false;
```

Comment:
Same

Reply:
I didn't not merge all if states into a huge if a and b or c and d or e and f. Since that would make it unreadable.